specification/utxo-fastsync-chip-20210625.md

```diff
@@ -94,20 +94,17 @@ The serialized UTXO set is the set of byte sequences constructed by serializing
 | Byte Count | Name | Description | Endian |
 | --- | --- | --- | --- |
 | 32 | transactionHash | The hash ("txid") of the transaction. | Little |
-| 4 | outputIndex | The index of the output within the transaction. | Little |
-| 4 | height, isCoinbase | The most significant bit is set if the UTXO is a coinbase output.  The remaining 31 bits** represent the block height. | Little |
-| 8 | value | The amount, in satoshis. | Little |
-| 1-4 | lockingScriptByteCount | The number of bytes in the locking script (as a compact variable length integer**). | Big |
-| ? | lockingScript | The locking script ("scriptPubKey", "pk_script"). | Big |
+| 1-5 | outputIndex | The index of the output within the transaction (as a compact variable length integer**). | Little |
+| 4 | height, isCoinbase | This is serialized as an unsigned 32-bit integer. The least significant bit is set if the UTXO is a coinbase output.  The remaining high-order 31 bits represent the block height (shifted over by 1 bit). | Little |
+| 1-9 | value | The amount, in satoshis, negative values are disallowed, so the useful range is 0 - 2^63-1. Serialized as a compact variable length integer**. | Little |
+| 1-5 | lockingScriptByteCount | The number of bytes in the locking script (as a compact variable length integer**). | Little |
+| ? | lockingScript | The locking script ("scriptPubKey", "pk_script"). For token outputs, should include the preamble TOKEN_PREFIX + token data (`0xef`, etc..) | N/A |
 
-** The Van der Wansem proposal was ambiguous in regard to its definition of the locking script byte count and UTXO height.
-This proposal uses a 4-byte LE integer for the UTXO height/isCoinbase flag, and a 1 byte (up to 4) variable compact-length integer for its locking-script byte-count prefix.
-Considering there are over 55 million UTXOs in the current set, many of which could use 1 byte instead of 4 bytes to encode the lockingScript byte count, converting this field to a compact variable length integer reduces the current UTXO snapshot size by ~160 MB.
-This reduction is not unremarkable and poses little complexity to the format.
+** The Van der Wansem proposal was ambiguous in regard to its definition of the locking script byte count and UTXO height, and it used fixed-width 8-byte integers for the UTXO value.
+This proposal uses variable [compact variable-length integers](https://documentation.cash/protocol/formats/variable-length-integer) also known as `CompactSzie` in many places to save space within he UTXO format, as well as other places in this specification.
+Considering there are over 55 million UTXOs in the current set, opting to encode all integers as `CompactSize` ints (little endian) may save hundreds of MB of space. This reduction is not unremarkable and poses little complexity to the format.
```

Comment:
Minor nitpick, but "encode all integers as `CompactSize` ints" conflicts with the prior sentence and generally could be misleading, so it might be more clear to just say "some integers"?

Reply:
Yeah you are right -- this was from an earlier version where I wanted to CompactSize *all the things*. Will fix.

Reply:
```suggestion
Considering there are over 55 million UTXOs in the current set, opting to encode some integers as `CompactSize` ints (little endian) may save hundreds of MB of space. This reduction is not unremarkable and poses little complexity to the format.
```